dev/unix/install.sh.in

```diff
@@ -88,6 +88,33 @@ notion_create_tree() {
   mkdir -p "${INSTALL_DIR}"/tmp
 }
 
+# Safely handle an existing installation by removing it given user permission to
+# do so. If the user has no existing installation, this is a no-op.
+notion_check_existing_installation() {
+  local INSTALL_DIR="$(notion_install_dir)"
+  local NOTION_BIN="${INSTALL_DIR}/notion"
+
+  if [[ ! -z $INSTALL_DIR && -f $NOTION_BIN ]]; then
+    local PREV_NOTION_VERSION=`$NOTION_BIN version`
+    if [[ $PREV_NOTION_VERSION == 0.1* ]]; then
+      echo "        You have an earlier version of notion installed."
```

Comment:
How about something like:

```
    You have an earlier version of notion installed, and the version you are installing requires
    a fresh installation. (We will work to eventually eliminate the need for reinstallations as
    notion stabilizes! Thanks for bearing with us.)

    Remove existing installation and continue? [Y/n]
```

Reply:
Going that way was the other direction I thought of – seems good to me. I'll probably cap line length at 72 or something just to make sure it's readable in most terminal setups (unless you have a nice guide handy for auto wrapping it *with* indentation – that's the part I couldn't figure out).